core/src/main/java/com/sequenceiq/cloudbreak/service/cluster/HadoopConfigurationService.java

```diff
@@ -19,37 +25,109 @@
 public class HadoopConfigurationService {
 
     public static final String YARN_SITE = "yarn-site";
+    public static final String YARN_NM_LOCAL_DIRS = "yarn.nodemanager.local-dirs";
+    public static final String YARN_NM_LOG_DIRS = "yarn.nodemanager.log-dirs";
+    public static final String YARN_TIMELINE_LEVELDB_STORE = "yarn.timeline-service.leveldb-timeline-store.path";
+
     public static final String HDFS_SITE = "hdfs-site";
-    public static final String YARN_NODEMANAGER_LOCAL_DIRS = "yarn.nodemanager.local-dirs";
-    public static final String YARN_NODEMANAGER_LOG_DIRS = "yarn.nodemanager.log-dirs";
-    public static final String HDFS_DATANODE_DATA_DIRS = "dfs.datanode.data.dir";
+    public static final String HDFS_DN_DATA_DIRS = "dfs.datanode.data.dir";
+    public static final String HDFS_NN_CHECKPOINT_DIR = "dfs.namenode.checkpoint.dir";
+    public static final String HDFS_NN_NAME_DIR = "dfs.namenode.name.dir";
+    public static final String HDFS_JN_EDITS_DIR = "dfs.journalnode.edits.dir";
+
+    public static final String HBASE_SITE = "hbase-site";
+    public static final String HBASE_TMP_DIR = "hbase.tmp.dir";
+
+    public static final String STORM_SITE = "storm-site";
+    public static final String STORM_LOCAL_DIR = "storm.local.dir";
+
+    public static final String OOZIE_ENV = "oozie-env";
+    public static final String OOZIE_DATA_DIR = "oozie_data_dir";
+
+    public static final String ZOOKEEPER_CFG = "zoo.cfg";
+    public static final String ZOOKEEPER_DATA_DIR = "dataDir";
+
+    public static final String FALCON_STARTUP = "falcon-startup.properties";
+    public static final String FALCON_GRAPH_PATH = "*.falcon.graph.serialize.path";
+    public static final String FALCON_GRAPH_DIR = "*.falcon.graph.storage.directory";
+
+    private static Map<String, Map<String, List<String>>> requiredGlobalConfigs = new HashMap<>();
```

Comment:
No change is required, just I want to emphasise  that it can simplified a bit:
- For Map<K, List<V> cases I would use the Multimap from Guava, this is my personal preference, it looks simpler for me
- Introducing a classes for such complicated structure is also not a bad idea, it makes the code much more readable